### PR TITLE
Ros2 lights integration tests

### PR DESCRIPTION
**Pull request by pawelirh**
### Description

- Fixed mistake copyright
- Segregated test files between `unit` and `integration`
- Implemented integration tests for the `panther_lights` package 

### Requirements

- [X] Code style guidelines followed
- [X] Documentation updated

### Tests 🧪

- [X] Robot
- [ ] Container
- [ ] Simulation

## Summary by CodeRabbit

- **New Features**
	- Updated copyright notice to reflect new ownership.
	- Introduced a new option to enable or disable integration tests.
	- Added a new integration test suite for the `panther_lights` package, ensuring robust testing of LED control functionalities.
	- Introduced a utility for retrieving information about ROS 2 nodes.

- **Bug Fixes**
	- None

- **Documentation**
	- None

- **Refactor**
	- Restructured test file paths for better organization.

- **Chores**
	- Updated copyright year in the relevant files.

### CODE REVIEW

File: panther_lights/CMakeLists.txt [hunk 1]
```diff
@@ -221,6 +225,12 @@ if(BUILD_TESTING)
   target_link_libraries(${PROJECT_NAME}_test_lights_controller_node
                         lights_controller_node_component yaml-cpp)
 
+  # Integration tests
+  option(TEST_INTEGRATION "Run integration tests" OFF)
+  if(TEST_INTEGRATION)
+    add_ros_test(test/integration/panther_lights.test.py)
+  endif()
+
```
Comment (KmakD) [inline panther_lights/CMakeLists.txt:233]: should we move this outside BUILD_TESTING? This way we could run only integration tests on remote.
Unit test can be done on local machine and building them on built-in computer can take a lot of time.

File: panther_lights/test/integration/panther_lights.test.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            srv_name="hardware/led_control_enable",
+            callback=self.led_control_enable_cb,
+            qos_profile=rclpy.qos.qos_profile_services_default,
+        )
+
+        self.set_led_animation_client = self.test_node.create_client(
+            srv_type=SetLEDAnimation,
+            srv_name="lights/set_animation",
+            qos_profile=rclpy.qos.qos_profile_services_default,
+        )
```
Comment (KmakD) [inline panther_lights/test/integration/panther_lights.test.py]:
```suggestion
        self._test_node = rclpy.create_node("test_node")
        self._led_control_requested = None

        self._led_control_enable_srv = self.test_node.create_service(
            srv_type=SetBool,
            srv_name="hardware/led_control_enable",
            callback=self.led_control_enable_cb,
            qos_profile=rclpy.qos.qos_profile_services_default,
        )

        self._set_led_animation_client = self.test_node.create_client(
            srv_type=SetLEDAnimation,
            srv_name="lights/set_animation",
            qos_profile=rclpy.qos.qos_profile_services_default,
        )
```
Global variables should start with underscore

File: panther_lights/test/integration/panther_lights.test.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    def tearDown(self):
+        self.test_node.destroy_node()
+
+    def led_control_enable_cb(self, request, response):
```
Comment (KmakD) [inline panther_lights/test/integration/panther_lights.test.py]:
```suggestion
    def _led_control_enable_cb(self, request, response):
```
Private methods should start with underscore

File: panther_lights/test/integration/panther_lights.test.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    def request_error_animation(self):
+        request = SetLEDAnimation.Request()
+        request.animation = LEDAnimation(id=LEDAnimation.ERROR)
+        request.repeating = False
+
+        self.set_led_animation_client.wait_for_service(timeout_sec=1.0)
+        self.set_led_animation_client.call_async(request)
+
+    def test_msg_publishers(self):
```
Comment (KmakD) [inline panther_lights/test/integration/panther_lights.test.py:125]: Is there a simple way to check msg subscribers? Like check if lights_driver_node is subscribed to frame topics?

**Review by KmakD — CHANGES_REQUESTED** [review]

File: panther_lights/CMakeLists.txt (hunk 1 above)
Comment (pawelirh) [reply panther_lights/CMakeLists.txt:233]: As we discussed, it is the most appropriate behavior for now.

File: panther_lights/test/integration/panther_lights.test.py (hunk 2 above)
Comment (pawelirh) [reply panther_lights/test/integration/panther_lights.test.py]: Prefix added 👍🏼

File: panther_lights/test/integration/panther_lights.test.py (hunk 4 above)
Comment (pawelirh) [reply panther_lights/test/integration/panther_lights.test.py:125]: Added new functionality 👍🏼

File: panther_utils/panther_utils/integration_test_utils.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                current_section = section_map[line]
+            elif line and current_section is not None:
+                current_section.append(line.split(":")[0].strip())
+            else:
+                current_section = None
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing command: {e}")
+        print(f"stderr: {e.stderr}")
+
```
[the remark this answers is not in the document]
Comment (pawelirh) [reply panther_utils/panther_utils/integration_test_utils.py]: Changed error handling

**Review by KmakD — APPROVED** [review]

**Pull request merged**